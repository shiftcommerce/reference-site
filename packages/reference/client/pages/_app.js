import React from 'react'
import App, { Container } from 'next/app'
import { Config } from 'shift-react-components'
import { shiftNextConfig } from 'shift-next'

// Objects
import NextHead from '../objects/next-head'
import NextLink from '../objects/next-link'

// Next config
import getConfig from 'next/config'

// Stylesheet
import '../scss/main.scss'

const {
  publicRuntimeConfig: {
    ALGOLIA_API_KEY,
    ALGOLIA_APP_ID,
    ALGOLIA_RESULTS_PER_PAGE,
    ALGOLIA_INDEX_NAME,
    API_HOST_PROXY,
    STRIPE_API_KEY
  }
} = getConfig()

Config.set({
  Head: NextHead,
  Link: NextLink,
  stripeApiKey: STRIPE_API_KEY
})

shiftNextConfig.set({
  algoliaApiKey: ALGOLIA_API_KEY,
  algoliaAppId: ALGOLIA_APP_ID,
  algoliaIndexName: ALGOLIA_INDEX_NAME,
  algoliaResultsPerPage: ALGOLIA_RESULTS_PER_PAGE,
  apiHostProxy: API_HOST_PROXY,
  storeName: 'ShopGo'
})

class MyApp extends App {
  render () {
    const { Component, pageProps } = this.props
    return (
      <Container>
        <Component {...pageProps} />
      </Container>
    )
  }
}

export default MyApp
