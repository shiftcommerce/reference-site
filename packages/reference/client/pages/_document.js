// _document is only rendered on the server side and not on the client side
// Event handlers like onClick can't be added to this file

// ./pages/_document.js
import Document, { Html, Head, Main, NextScript } from 'next/document'

// Next config
import getConfig from 'next/config'

const { publicRuntimeConfig: { PAYPAL_CLIENT_ID } } = getConfig()

class MyDocument extends Document {
  static async getInitialProps (ctx) {
    const initialProps = await Document.getInitialProps(ctx)

    return { ...initialProps }
  }

  render () {
    const payPalClientID = PAYPAL_CLIENT_ID

    return (
      <Html>
        <Head>
          <script src='https://js.stripe.com/v3/' key='stripe' />
          { payPalClientID && <script src={`https://www.paypal.com/sdk/js?client-id=${payPalClientID}&currency=GBP&intent=authorize&commit=false&disable-funding=credit,sepa,card`} /> }
        </Head>
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    )
  }
}

export default MyDocument
