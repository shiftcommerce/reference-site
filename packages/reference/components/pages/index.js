import CategorySidebar from './categorySidebar'

const PageTemplates = {
  category_sidebar_template: CategorySidebar
}

export default PageTemplates
