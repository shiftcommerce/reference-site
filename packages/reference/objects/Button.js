// Libraries
import { Component } from 'react'
import classNames from 'classnames'

class Button extends Component {
  render () {
    let {
      status,
      label,
      size,
      disabled,
      className,
      ...otherProps
    } = this.props

    disabled = disabled || false
    status = status || 'primary'
    size = size || 'lrg'

    return (
      <button
        className={classNames(className, 'o-button', `o-button--${status}`, `o-button--${size}`, {'o-button--disabled': disabled})}
        tabIndex='0'
        role='button'
        disabled={disabled}
        {...otherProps}
      >
        { label }
      </button>
    )
  }
}

export default Button
