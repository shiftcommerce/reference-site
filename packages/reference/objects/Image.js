import { Component } from 'react'
import classNames from 'classnames'

class Image extends Component {
  styles () {
    let basicStyles = {
      height: this.props.height,
      width: this.props.width
    }

    if (basicStyles.height < 200) {
      return { ...basicStyles, backgroundSize: ((basicStyles.height * 19) / 20) }
    } else if (basicStyles.width < 200) {
      return { ...basicStyles, backgroundSize: ((basicStyles.width * 19) / 20) }
    } else {
      return { ...basicStyles, backgroundSize: 150 }
    }
  }

  render () {
    let {
      src,
      className
    } = this.props

    if (src) {
      return <img {...this.props} />
    } else {
      return <div className={classNames('c-dummy_image', className)} style={this.styles()} />
    }
  }
}

export default Image
