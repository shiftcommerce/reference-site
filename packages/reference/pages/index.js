import { Component } from 'react'

// components
import Layout from '../components/Layout'

class Home extends Component {
  render () {
    return <Layout>
      <div>Welcome To Shift Commerce Front End React</div>
    </Layout>
  }
}

export default Home
