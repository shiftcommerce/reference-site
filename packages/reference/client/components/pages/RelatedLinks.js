// Libraries
import { Component } from 'react'
import { connect } from 'react-redux'
import Link from 'next/link'

// Objects
import Button from '../../objects/Button'

export class RelatedLinks extends Component {
  linkButton (linkData, element) {
    return (
      <div className='c-links__grid-button'>
        <Link href={`/slug?slug=${linkData[`link${element}_url`][0].canonical_path}`} as={linkData[`link${element}_url`][0].canonical_path}>
          <Button className='c-links__grid-button-icon' label={linkData[`link${element}_text`]} status='warning' size='lrg' aria-label={linkData[`link${element}_text`]} />
        </Link>
      </div>
    )
  }

  render () {
    const { links } = this.props
    const elements = [1, 2, 3]

    return (
      <div className='c-links'>
        <div className='c-links__text'>
          <h1 className='c-links__text-title'>
            {links.title}
          </h1>
        </div>
        <div className='c-links__grid'>
          { elements.map((element, index) => {
            if (links[`link${element}_url`]) {
              return (
                <div key={index}>
                  {this.linkButton(links, element)}
                </div>
              )
            }
          }) }
        </div>
      </div>
    )
  }
}

export default connect()(RelatedLinks)
