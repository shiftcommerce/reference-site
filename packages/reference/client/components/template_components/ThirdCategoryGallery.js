// Libraries
import { Component } from 'react'
import { connect } from 'react-redux'
import Link from 'next/link'

// Objects
import Button from '../../objects/Button'
import Image from '../../objects/Image'

export class ThirdCategoryGallery extends Component {
  thirdComponent (thirdData, element) {
    return (
      <div className='c-third'>
        <Image className='c-third__image' src={thirdData[`box${element}_image`][0].s3_url} />
        <h1 className='c-third__title'>{thirdData[`box${element}_title`]}</h1>
        <div className='c-third__text'>
          <p>{thirdData[`box${element}_text`]}</p>
        </div>
        <div className='c-third__button'>
          <Link href={`/slug?slug=${thirdData[`box${element}_link_url`][0].canonical_path}`} as={thirdData[`box${element}_link_url`][0].canonical_path}>
            <Button className='c-third__button-icon' label={thirdData[`box${element}_link_text`]} status='warning' size='lrg' aria-label={thirdData[`box${element}_link_text`]} />
          </Link>
        </div>
      </div>
    )
  }

  render () {
    const { componentData } = this.props
    const elements = [1, 2, 3]

    return (
      <div className='c-third-gallery'>
        { elements.map((element, index) => {
          return (
            <div key={index}>
              {this.thirdComponent(componentData, element)}
            </div>
          )
        }) }
      </div>
    )
  }
}

export default connect()(ThirdCategoryGallery)
