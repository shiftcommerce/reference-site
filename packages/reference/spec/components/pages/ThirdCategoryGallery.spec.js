// Components
import { ThirdCategoryGallery } from '../../../components/pages/ThirdCategoryGallery'

// Objects
import Button from '../../../objects/Button'
import Image from '../../../objects/Image'

// Fixtures
import thirdCategoryData from '../../fixtures/thirdCategoryGallery.fixture'

test('renders the full ThirdCategoryGallery component', () => {
  // Arrange
  const emptyFunction = () => {}

  // Act
  const wrapper = mount(
    <ThirdCategoryGallery third={thirdCategoryData} />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText(thirdCategoryData.box1_title)
  expect(wrapper).toIncludeText(thirdCategoryData.box2_title)
  expect(wrapper).toIncludeText(thirdCategoryData.box3_title)
  expect(wrapper).toIncludeText(thirdCategoryData.box1_text)
  expect(wrapper).toIncludeText(thirdCategoryData.box2_text)
  expect(wrapper).toIncludeText(thirdCategoryData.box3_text)
  expect(wrapper).toContainReact(<Image className='c-third__image' src={thirdCategoryData.box1_image[0].s3_url} />)
  expect(wrapper).toContainReact(<Image className='c-third__image' src={thirdCategoryData.box2_image[0].s3_url} />)
  expect(wrapper).toContainReact(<Image className='c-third__image' src={thirdCategoryData.box3_image[0].s3_url} />)
  expect(wrapper.find('Button').at(0)).toMatchElement(<Button className='c-third__button-icon' label={thirdCategoryData.box1_link_text} status='warning' size='lrg' aria-label={thirdCategoryData.box1_link_text} onClick={emptyFunction} />)
  expect(wrapper.find('Button').at(1)).toMatchElement(<Button className='c-third__button-icon' label={thirdCategoryData.box2_link_text} status='warning' size='lrg' aria-label={thirdCategoryData.box2_link_text} onClick={emptyFunction} />)
  expect(wrapper.find('Button').at(2)).toMatchElement(<Button className='c-third__button-icon' label={thirdCategoryData.box3_link_text} status='warning' size='lrg' aria-label={thirdCategoryData.box3_link_text} onClick={emptyFunction} />)
})
