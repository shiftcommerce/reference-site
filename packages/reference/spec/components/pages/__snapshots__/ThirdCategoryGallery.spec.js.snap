// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the full ThirdCategoryGallery component 1`] = `
<ThirdCategoryGallery
  third={
    Object {
      "box1_image": Array [
        Object {
          "asset_file": "1513676837.1465595-Week-44---Sale_03.jpg",
          "canonical_path": "https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157793/1513676837.1465595-Week-44---Sale_03.jpg",
          "created_at": "2017-12-19T09:47:17Z",
          "file_content_content_type": "image/jpeg",
          "file_content_filename": "Week-44---Sale_03.jpg",
          "file_content_size": 99662,
          "id": "157793",
          "image_height": 910,
          "image_width": 800,
          "reference": "74a6e169-9734-4d7b-884d-f8ae9ff90b68",
          "s3_url": "https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157793/1513676837.1465595-Week-44---Sale_03.jpg",
          "updated_at": "2017-12-19T09:47:17Z",
        },
      ],
      "box1_link_text": "shop womens sportswear",
      "box1_link_url": Array [
        Object {
          "canonical_path": "/womens/shop-by-category/sportswear",
          "id": "/womens/shop-by-category/sportswear",
          "path": "/womens/shop-by-category/sportswear",
          "url": "/womens/shop-by-category/sportswear",
        },
      ],
      "box1_text": "Who doesn’t love a New Year’s cliché? Working out and getting fit is always a big deal in January, so we why not look good doing it – say hello to our latest Souluxe collection.",
      "box1_title": "GET GYMSPIRED",
      "box2_image": Array [
        Object {
          "asset_file": "1513676837.280586-Week-44---Sale_05.jpg",
          "canonical_path": "https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157794/1513676837.280586-Week-44---Sale_05.jpg",
          "created_at": "2017-12-19T09:47:17Z",
          "file_content_content_type": "image/jpeg",
          "file_content_filename": "Week-44---Sale_05.jpg",
          "file_content_size": 113653,
          "id": "157794",
          "image_height": 910,
          "image_width": 800,
          "reference": "30175cb0-2fd3-4608-92d1-57bf42e87bbd",
          "s3_url": "https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157794/1513676837.280586-Week-44---Sale_05.jpg",
          "updated_at": "2017-12-19T09:47:17Z",
        },
      ],
      "box2_link_text": "shop mens sportswear",
      "box2_link_url": Array [
        Object {
          "canonical_path": "/mens/categories/sportswear",
          "id": "/mens/categories/sportswear",
          "path": "/mens/categories/sportswear",
          "url": "/mens/categories/sportswear",
        },
      ],
      "box2_text": "Whether you’re new to working out or a total pro, our Souluxe collection is bursting full great gear – meaning you’ll look and feel great. Run, don’t walk for these beauties.",
      "box2_title": "BETTER, FASTER, STRONGER.",
      "box3_image": Array [
        Object {
          "asset_file": "1513676837.490249-Week-44---Sale_07.jpg",
          "canonical_path": "https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157795/1513676837.490249-Week-44---Sale_07.jpg",
          "created_at": "2017-12-19T09:47:17Z",
          "file_content_content_type": "image/jpeg",
          "file_content_filename": "Week-44---Sale_07.jpg",
          "file_content_size": 92114,
          "id": "157795",
          "image_height": 910,
          "image_width": 800,
          "reference": "0ccfb1ad-24a1-4fa3-93e3-9d1503547578d",
          "s3_url": "https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157795/1513676837.490249-Week-44---Sale_07.jpg",
          "updated_at": "2017-12-19T09:47:17Z",
        },
      ],
      "box3_link_text": "shop kids sportswear",
      "box3_link_url": Array [
        Object {
          "canonical_path": "/kids-clothing/kidswear/sportswear",
          "id": "/kids-clothing/kidswear/sportswear",
          "path": "/kids-clothing/kidswear/sportswear",
          "url": "/kids-clothing/kidswear/sportswear",
        },
      ],
      "box3_text": "If she loves to express herself through movement our latest looks from Souluxe will leave her with hearts in her eyes. Full of cute co-ords and bold prints, she’ll feel like the star of the show.",
      "box3_title": "MINI MOVES",
      "id": "15_3",
      "reference": "thirds_style1",
    }
  }
>
  <div
    className="c-third-gallery"
  >
    <div
      key="0"
    >
      <div
        className="c-third"
      >
        <Image
          className="c-third__image"
          src="https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157793/1513676837.1465595-Week-44---Sale_03.jpg"
        >
          <div
            className="c-third__image"
          >
            <LazyLoad
              height={200}
              offset={0}
              once={false}
              overflow={false}
              resize={false}
              scroll={true}
              unmountIfInvisible={false}
            >
              <div
                className="lazyload-placeholder"
                style={
                  Object {
                    "height": 200,
                  }
                }
              />
            </LazyLoad>
          </div>
        </Image>
        <h1
          className="c-third__title"
        >
          GET GYMSPIRED
        </h1>
        <div
          className="c-third__text"
        >
          <p>
            Who doesn’t love a New Year’s cliché? Working out and getting fit is always a big deal in January, so we why not look good doing it – say hello to our latest Souluxe collection.
          </p>
        </div>
        <div
          className="c-third__button"
        >
          <Link
            as="/womens/shop-by-category/sportswear"
            href="/slug?slug=/womens/shop-by-category/sportswear"
          >
            <Button
              aria-label="shop womens sportswear"
              className="c-third__button-icon"
              label="shop womens sportswear"
              onClick={[Function]}
              size="lrg"
              status="warning"
            >
              <button
                aria-label="shop womens sportswear"
                className="c-third__button-icon o-button o-button--warning o-button--lrg"
                disabled={false}
                onClick={[Function]}
                role="button"
                tabIndex="0"
              >
                shop womens sportswear
              </button>
            </Button>
          </Link>
        </div>
      </div>
    </div>
    <div
      key="1"
    >
      <div
        className="c-third"
      >
        <Image
          className="c-third__image"
          src="https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157794/1513676837.280586-Week-44---Sale_05.jpg"
        >
          <div
            className="c-third__image"
          >
            <LazyLoad
              height={200}
              offset={0}
              once={false}
              overflow={false}
              resize={false}
              scroll={true}
              unmountIfInvisible={false}
            >
              <div
                className="lazyload-placeholder"
                style={
                  Object {
                    "height": 200,
                  }
                }
              />
            </LazyLoad>
          </div>
        </Image>
        <h1
          className="c-third__title"
        >
          BETTER, FASTER, STRONGER.
        </h1>
        <div
          className="c-third__text"
        >
          <p>
            Whether you’re new to working out or a total pro, our Souluxe collection is bursting full great gear – meaning you’ll look and feel great. Run, don’t walk for these beauties.
          </p>
        </div>
        <div
          className="c-third__button"
        >
          <Link
            as="/mens/categories/sportswear"
            href="/slug?slug=/mens/categories/sportswear"
          >
            <Button
              aria-label="shop mens sportswear"
              className="c-third__button-icon"
              label="shop mens sportswear"
              onClick={[Function]}
              size="lrg"
              status="warning"
            >
              <button
                aria-label="shop mens sportswear"
                className="c-third__button-icon o-button o-button--warning o-button--lrg"
                disabled={false}
                onClick={[Function]}
                role="button"
                tabIndex="0"
              >
                shop mens sportswear
              </button>
            </Button>
          </Link>
        </div>
      </div>
    </div>
    <div
      key="2"
    >
      <div
        className="c-third"
      >
        <Image
          className="c-third__image"
          src="https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157795/1513676837.490249-Week-44---Sale_07.jpg"
        >
          <div
            className="c-third__image"
          >
            <LazyLoad
              height={200}
              offset={0}
              once={false}
              overflow={false}
              resize={false}
              scroll={true}
              unmountIfInvisible={false}
            >
              <div
                className="lazyload-placeholder"
                style={
                  Object {
                    "height": 200,
                  }
                }
              />
            </LazyLoad>
          </div>
        </Image>
        <h1
          className="c-third__title"
        >
          MINI MOVES
        </h1>
        <div
          className="c-third__text"
        >
          <p>
            If she loves to express herself through movement our latest looks from Souluxe will leave her with hearts in her eyes. Full of cute co-ords and bold prints, she’ll feel like the star of the show.
          </p>
        </div>
        <div
          className="c-third__button"
        >
          <Link
            as="/kids-clothing/kidswear/sportswear"
            href="/slug?slug=/kids-clothing/kidswear/sportswear"
          >
            <Button
              aria-label="shop kids sportswear"
              className="c-third__button-icon"
              label="shop kids sportswear"
              onClick={[Function]}
              size="lrg"
              status="warning"
            >
              <button
                aria-label="shop kids sportswear"
                className="c-third__button-icon o-button o-button--warning o-button--lrg"
                disabled={false}
                onClick={[Function]}
                role="button"
                tabIndex="0"
              >
                shop kids sportswear
              </button>
            </Button>
          </Link>
        </div>
      </div>
    </div>
  </div>
</ThirdCategoryGallery>
`;
